open_discussions/authentication.py

```diff
@@ -0,0 +1,92 @@
+"""Custom authentication for DRF"""
+import logging
+from base64 import b64decode, b64encode
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core import signing
+from rest_framework.authentication import BaseAuthentication
+
+User = get_user_model()
+
+HEADER_PREFIX = "Token "
+HEADER_PREFIX_LENGTH = len(HEADER_PREFIX)
+
+logger = logging.getLogger()
+
+
+def get_encoded_and_signed_subscription_token(user):
+    """
+    Returns a signed and encoded token for subscription authentication_classes
+
+    Args:
+        user (User): user to generate the token for
+
+    Returns:
+        str: a signed and encoded token for subscription authentication_classes
+    """
+    signer = signing.TimestampSigner()
+    signed_value = signer.sign(user.username)
+    return b64encode(signed_value.encode('utf-8')).decode('ascii')
```

Comment:
are we doing this `.encode` / `.decode` juggling because the username can have UTF-8 characters in it?

Reply:
(that is to say, UTF-8 characters which are not also ASCII characters)

Reply:
In Python 3, `b64encode` requires a bytes-like object instead of a string and returns a bytes-like object so we have to then decode it. I don't believe django signing returns any values outside of the ascii range but utf-8 is certainly safer. I'll update the code so the encode/decode is symmetrical though.